### PR TITLE
add context to warning in mutate()

### PR DESCRIPTION
**Pull request by romainfrancois**
Handling warnings like errors, and adding some information about when the warning occurred gives: 

``` r
library(tidyverse)
foo_warn <- function(x, sym){
  if(any(x > 3)){
    warning(paste(sym, "has an implausible value"))
  }
  x
}
iris %>% 
  mutate(
    across(contains("Sepal"), ~foo_warn(.x, cur_column()))
  ) %>% 
  head()
#> Warning: Problem with `mutate()` input `..1`.
#> x Sepal.Length has an implausible value
#> ℹ Input `..1` is `across(contains("Sepal"), ~foo_warn(.x, cur_column()))`.
#>   Sepal.Length Sepal.Width Petal.Length Petal.Width Species
#> 1          5.1         3.5          1.4         0.2  setosa
#> 2          4.9         3.0          1.4         0.2  setosa
#> 3          4.7         3.2          1.3         0.2  setosa
#> 4          4.6         3.1          1.5         0.2  setosa
#> 5          5.0         3.6          1.4         0.2  setosa
#> 6          5.4         3.9          1.7         0.4  setosa
```

<sup>Created on 2020-06-09 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

I'll apply this to other functions if this is what you meant @hadley ?

### CODE REVIEW

**Comment by romainfrancois** [comment]
closes #5313

File: R/conditions.R [hunk 1]
```diff
@@ -57,7 +57,7 @@ stop_dplyr <- function(.index, dots, fn, problem, ..., .dot_data = FALSE, .show_
     if(.show_group_details) cnd_bullet_cur_group_label(),
     .envir = envir
   )
-  abort(
+  .abort(
```
Comment (lionel-) [inline R/conditions.R:60]: Should be called `.signal`.

Comment (hadley) [reply R/conditions.R:60]: And maybe call the whole function something like `with_group_context()`?

Comment (lionel-) [reply R/conditions.R:60]: I think the current name `stop_dplyr()` is more informative at the call sites? It's more explicit that this is a stop-like function.

**Comment by romainfrancois** [comment]
What about something like:

```
warn_dplyr <- stop_dplyr
formals(warn_dplyr)$.signal <- sym("warn")
```

**Pull request merged**